nifi-nar-bundles/nifi-opentelemetry-bundle/nifi-opentelemetry-processors/src/main/java/org/apache/nifi/processors/opentelemetry/ListenOTLP.java

```diff
@@ -0,0 +1,241 @@
+            .build();
+
+    static final PropertyDescriptor WORKER_THREADS = new PropertyDescriptor.Builder()
+            .name("Worker Threads")
+            .displayName("Worker Threads")
+            .description("Number of threads responsible for decoding and queuing incoming OTLP Export Service Requests")
+            .required(true)
+            .defaultValue("2")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .build();
+
+    static final PropertyDescriptor QUEUE_CAPACITY = new PropertyDescriptor.Builder()
+            .name("Queue Capacity")
+            .displayName("Queue Capacity")
+            .description("Maximum number of OTLP request resource elements that can be received and queued")
+            .required(true)
+            .defaultValue("10000")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .displayName("Batch Size")
+            .description("Maximum number of OTLP request resource elements included in each FlowFile produced")
+            .required(true)
+            .defaultValue("1000")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .build();
+
+    static final Relationship SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Export Service Requests containing OTLP Telemetry")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.singleton(SUCCESS);
+
+    private static final List<PropertyDescriptor> DESCRIPTORS = List.of(
+            ADDRESS,
+            PORT,
+            SSL_CONTEXT_SERVICE,
+            CLIENT_AUTHENTICATION,
+            WORKER_THREADS,
+            QUEUE_CAPACITY,
+            BATCH_SIZE
+    );
+
+    private static final String TRANSIT_URI_FORMAT = "https://%s:%d";
+
+    private Iterator<RequestCallback> requestCallbackProvider;
+
+    private EventServer server;
+
+    @Override
+    public final Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws UnknownHostException {
+        final EventServerFactory eventServerFactory = createEventServerFactory(context);
+        server = eventServerFactory.getEventServer();
+    }
+
+    @OnStopped
+    public void onStopped() {
+        if (server == null) {
+            getLogger().info("Server not running");
```

Comment:
This line seems a little strange.  It seems the only time the server could be null is if this processor was stopped so if that happens this method is already done/called.  And even if this could happen the output doesn't seem useful to the user as there is no meaning they can really derive or action they can take.  Was this from earlier testing and left in or am I not considering some case?

Reply:
Thanks for the note and question. Yes, it was from some early iterations, as it highlights the server status in some edge cases when using the mock TestRunner with different arguments. However, I agree that in normal operation, it should not happen, it there is nothing to do anyway, so I will simplify the logic and avoid the log.